### PR TITLE
Port changes from arangodb main repo

### PR DESCRIPTION
**Pull request by graetzer**
- Properly handle idle connection timeout
- support interpreting "keep-alive" header with timeout
- only cancel in-flight requests, not all queued requests

### CODE REVIEW

File: include/fuerte/connection.h [hunk 1]
```diff
@@ -80,7 +86,7 @@ class Connection : public std::enable_shared_from_this<Connection> {
   
   /// @brief Return the number of bytes that still need to be transmitted
   std::size_t bytesToSend() const {
-    return _bytesToSend.load(std::memory_order_acquire);
+    return _bytesToSend.load(std::memory_order_relaxed);
```
Comment (dhly-etc) [inline include/fuerte/connection.h:89]: This doesn't seem particularly safe if the value is used for control flow. Is it gated behind something else with stronger synchronization?

File: src/HttpConnection.h [hunk 2]
```diff
@@ -60,91 +57,80 @@ class HttpConnection final : public fuerte::Connection {
...
   
-  /// is loop active
-  std::atomic<uint32_t> _numQueued;
-  std::atomic<bool> _active;
+ private:
   
   /// elements to send out
   boost::lockfree::queue<fuerte::v1::http::RequestItem*,
-    boost::lockfree::capacity<1024>> _queue;
+  boost::lockfree::capacity<1024>> _queue;
```
Comment (dhly-etc) [inline src/HttpConnection.h:105]:
```suggestion
                         boost::lockfree::capacity<1024>> _queue;
```

File: src/VstConnection.h [hunk 3]
```diff
@@ -128,24 +108,14 @@ class VstConnection final : public Connection {
...
-  Socket<ST> _protocol;
-  
-  /// @brief timer to handle connection / request timeouts
-  asio_ns::steady_timer _timeout;
-  
-  /// @brief is the connection established
-  std::atomic<Connection::State> _state;
+  /// elements to send out
+  boost::lockfree::queue<vst::RequestItem*,
+  boost::lockfree::capacity<1024>> _writeQueue;
```
Comment (dhly-etc) [inline src/VstConnection.h:113]:
```suggestion
                         boost::lockfree::capacity<1024>> _writeQueue;
```

File: src/HttpConnection.cpp [hunk 4]
```diff
@@ -34,118 +34,155 @@
...
   // initialize http parsing code
   http_parser_settings_init(&_parserSettings);
-  _parserSettings.on_message_begin = ::on_message_began;
-  _parserSettings.on_status = ::on_status;
-  _parserSettings.on_header_field = ::on_header_field;
-  _parserSettings.on_header_value = ::on_header_value;
-  _parserSettings.on_headers_complete = ::on_header_complete;
-  _parserSettings.on_body = ::on_body;
-  _parserSettings.on_message_complete = ::on_message_complete;
+  _parserSettings.on_message_begin = &on_message_began;
```
Comment (dhly-etc) [inline src/HttpConnection.cpp:154]: Maybe make `began` to `begin` so these match?

**Review by dhly-etc — CHANGES_REQUESTED** [review]

File: include/fuerte/connection.h (hunk 1 above)
Comment (graetzer) [reply include/fuerte/connection.h:89]: it's just a statistic

**Pull request closed without merging**